torch_np/_detail/_reductions.py

```diff
@@ -371,36 +367,73 @@ def average_weights(a_tensor, axis, w_tensor, keepdims=False):
     return result, denominator
 
 
-def quantile(a_tensor, q_tensor, axis, method, keepdims=False):
-
-    if (0 > q_tensor).any() or (q_tensor > 1).any():
-        raise ValueError("Quantiles must be in range [0, 1], got %s" % q_tensor)
-
-    if not a_tensor.dtype.is_floating_point:
+def quantile(
+    a,
+    q,
+    axis,
+    overwrite_input,
+    method,
+    keepdims=False,
+    interpolation=None,
+):
+    if overwrite_input:
+        # raise NotImplementedError("overwrite_input in quantile not implemented.")
+        # NumPy documents that `overwrite_input` MAY modify inputs:
+        # https://numpy.org/doc/stable/reference/generated/numpy.percentile.html#numpy-percentile
+        # Here we choose to work out-of-place because why not.
+        pass
+
+    if interpolation is not None:
+        raise ValueError("'interpolation' argument is deprecated; use 'method' instead")
+
+    if (0 > q).any() or (q > 1).any():
+        raise ValueError("Quantiles must be in range [0, 1], got %s" % q)
```

Comment:
I missed this one. `any()` synchronises and it creates a graph break in TorchDynamo. Let's leave it for now, but we may need to remove these checks in the future.

For now, let's try to use it as little as possible, so let's prefer `((0 > q) | (q > 1)).any()`

Reply:
Since this function delegates to `torch.quantile` below,  and `torch.quantile` has a similar check, I just removed this check here. The only price to pay is `assert_raises((RuntimeError, ValueError), ...)` in place of numpy's `assert_raises(ValueError, ...)` --- and we're using this in multiple places anyway.